backend/app/api/src/helpers/UitpasNumberValidator.ts

```diff
@@ -0,0 +1,114 @@
+                    'UiTPAS is niet volledig geconfigureerd. Neem contact op met de beheerder.',
+                ),
+            });
+        }
+        const access_token = await UitpasTokenRepository.getAccessTokenFor(); // for nothing, means for platform
+        const baseUrl = 'https://api-test.uitpas.be'; // TO DO: Use the URL from environment variables
+
+        const url = `${baseUrl}/passes/${uitpasNumber}`;
+        const myHeaders = new Headers();
+        myHeaders.append('Authorization', 'Bearer ' + access_token);
+        const requestOptions = {
+            method: 'GET',
+            headers: myHeaders,
+        };
+
+        const response = await fetch(url, requestOptions).catch((error) => {
+            // Handle network errors
+            throw new SimpleError({
+                code: 'uitpas_unreachable',
+                message: `Error retrieving pass by UiTPAS number: ${error.message}`,
+                human: $t(
+                    `De verbinding met UiTPAS is mislukt, dus konden we je UiTPAS-nummer niet valideren. Probeer het later opnieuw.`,
+                ),
+            });
+        });
+        if (!response.ok) {
+            // Handle non-200 responses by attempting to parse the response to get a message from UiTPAS API
+            const json = await response.json().catch(() => {}); // Ignore JSON parsing errors -> we will throw a generic error below
+            if (
+                json
+                && json.endUserMessage
+                && json.endUserMessage.nl
+                && typeof json.endUserMessage.nl === 'string'
+                && json.title
+                && typeof json.title === 'string'
+            ) {
+                // If the response contains an endUserMessage, throw an error with that message
+                throw new SimpleError({
+                    code: 'error_retrieving_pass_by_uitpas_number',
+                    message: `UiTPAS API returned an error: ${json.title}`,
+                    human: json.endUserMessage,
+                });
+            }
+            throw new SimpleError({
+                code: 'bad_response_when_fetching_uitpas_number',
+                message: `Unsuccessful response from UiTPAS API: ${response.statusText}`,
+                human: $t(
+                    `Er is een fout opgetreden bij het ophalen van je UiTPAS. Probeer het later opnieuw.`,
+                ),
+            });
+        }
+
+        const json = await response.json().catch((error) => {
+            // Handle JSON parsing errors
+            throw new SimpleError({
+                code: 'invalid_json_response',
+                message: `Invalid JSON response from UiTPAS API: ${error.message}`,
+                human: $t(
+                    `Er is een fout opgetreden bij het verwerken van de UiTPAS-gegevens. Probeer het later opnieuw.`,
+                ),
+            });
+        });
+        if (!json || !json.socialTariff || !json.socialTariff.status) {
```

Comment:
Dit is niet 100% veilig omdat json of json.socialTariff in principe ook bv. een nummer of string kan zijn, dan ga hier .status op socialTariff callen, wat een error zal gooien. Je moet nog checken of socialTariff een object is, en of json een object is (kan ook een array of iets anders zijn). Let op dat null ook een object is in javascript.

Kleine hint om ervoor te zorgen dat TypeScript je dit soort zaken meldt is hierboven iets aan te passen:

```const json = await response.json()```
naar
```const json: unknown = await response.json()```

De eerste is automatisch een 'any' omdat dit zo in de type definitions staat van de json methode. TypeScript is automatisch minder streng voor any dan voor unknown.